work_queue/src/work_queue.c

```diff
@@ -7543,6 +7549,25 @@ static void write_transaction_worker_resources(struct work_queue *q, struct work
 }
 
 
+static void write_transaction_transfer(struct work_queue *q, struct work_queue_worker *w, struct work_queue_task *t, size_t sizeinmb, int walltime, char* filename, int type){
```

Comment:
Note that size and walltime come in bytes, and microseconds, respectively.  I'd suggest changing the parameters to:
```C
..., size_t size_in_bytes, int time_in_usecs, work_queue_file_type_t type) {
```

and convert to MB and seconds as needed. Use "%f" when printing the numbers below.

Reply:
For memory, divide by `((double) MEGABYTE)`.  For time, divide by `((double) USECOND)`, and use "%f" in buffer_print for both.